### ISSUE TITLE
wav2vec2_with_lm

### ISSUE THREAD

**Issue by liroda** [huggingface/transformers#15236]
I refer to the example and run kenlm on the Chinese test set, it is ok ,but the language model weight, no matter how I adjust it, the result remains the same, this is my debugging code

 # dict
    vocabfile = os.path.join(model_path,'vocab.json')
    with open(vocabfile,'r',encoding="utf-8") as f:
        vocab_dict = json.load(f)

    labels = list(vocab_dict.keys())
    # kenlm deocder 

    decoder = build_ctcdecoder(
            labels,
            kenlm_model_path=kenlm_path,
            alpha = alpha,
            beta = beta
            )
    # processor_with_lm 

    processor_with_lm = Wav2Vec2ProcessorWithLM(
             feature_extractor=processor.feature_extractor,
             tokenizer=processor.tokenizer,
             decoder=decoder
             )

    # decode with lm 
    text_tmp = processor_with_lm.batch_decode(logits.numpy()).text[0].split('<s>')
    text_lm = ''.join(text_tmp)
    print ("lm text:{}".format(text_lm))

i  adjust   alpha  beta , how can i do  ？

**Comment by olafthiele** [comment]
You can run a hyperparameter search by inferencing a test set with different alpha and beta values. But if you have a small language model there might not be a difference. Maybe use a larger language model or larger test set.

**Comment by qinyuenlp** [comment]
> I refer to the example and run kenlm on the Chinese test set, it is ok ,but the language model weight, no matter how I adjust it, the result remains the same, this is my debugging code
> 
> # dict
> ```
> vocabfile = os.path.join(model_path,'vocab.json')
> with open(vocabfile,'r',encoding="utf-8") as f:
>     vocab_dict = json.load(f)
> 
> labels = list(vocab_dict.keys())
> # kenlm deocder 
> 
> decoder = build_ctcdecoder(
>         labels,
>         kenlm_model_path=kenlm_path,
>         alpha = alpha,
>         beta = beta
>         )
> # processor_with_lm 
> 
> processor_with_lm = Wav2Vec2ProcessorWithLM(
>          feature_extractor=processor.feature_extractor,
>          tokenizer=processor.tokenizer,
>          decoder=decoder
>          )
> 
> # decode with lm 
> text_tmp = processor_with_lm.batch_decode(logits.numpy()).text[0].split('<s>')
> text_lm = ''.join(text_tmp)
> print ("lm text:{}".format(text_lm))
> ```
> 
> i adjust alpha beta , how can i do ？

Check your kenlm model is char-level or word-level. To wav2vec2_with_lm, it bases `pyctcdecode`, which only support word-level kenlm. If you need to use char-level language model, you need to tune codes of `pyctcdecode`.
